### ISSUE TITLE
Please get familiar with modern ffmpeg use, and pitfalls of current settings.

### ISSUE THREAD

**Issue by YellowOnion** [caydey/ffshare#27]
Currently the settings make far too many assumptions and are extremely brittle.

I'm pretty sure the current defaults settings are outright wrong, because I can't even get any output at the moment, and the use of some commands don't look correct for a casual ffmpeg user myself.

1. Scaling is borked
   1. H.264 requires divisible by 2 resolutions, that means that the `-vf scale=-1:720` command just won't work 100% of the time especially for large downscaling, like with portrait videos (405x720)
   1. This command doesn't support portrait video recordings nicely, you should end up with 720x1280 videos.
   1. don't use `format` as a separate filter chain, there's literally an error saying it will be ignored.
   2. Because ffmpeg applies the display matrix during output scaling (portrait video from a phone is still 16:9 with a metadata telling the player to apply a rotation), it's probably better to use a high quality upscaler like lanczos to stop the chroma sub sampling getting worse, and it's not that expensive CPU wise.
   3. the full filter chain should probably look like this and should cover most common use cases and give sane output [[doc](https://ffmpeg.org/ffmpeg-filters.html#scale-1)]
 
   ```
   -vf scale=720:720:force_original_aspect_ratio=increase:force_divisible_by=2:flags=lanczos:out_range=tv,format=yuv420p,setsar=1/1
   ``` 

5. use libx264 explicitly, not implicitly, `-c:v libx264` don't relying on ffmpeg keeping the "h264" defaulting to libx264, especially when using codec specific settings like `-crf`, I get errors about incompatible muxer stuff, which is probably related to this.

6. Don't ruin my audio, just issue `-c:a  copy`, transcording *always* hurts quality, even increasing the bitrate, and is a waste of precious battery life, this should be the default when input audio AAC, and we don't want lower bitrate.

5.  mp4 is a shit output format, put [`-movflags faststart`](https://ffmpeg.org/ffmpeg-formats.html#Options-13) to make it slightly less shit.

7. expose the x264 presets as a menu option / profile, the x2 speed increase for settings like [`veryfast`](https://dev.beandog.org/x264_preset_reference.html) is definitely worth in in some situations
8. Please let us use custom settings, especially for x264, I love the [denoiser](https://en.wikibooks.org/wiki/MeGUI/x264_Settings#nr) but I know it's not everyones cup of tea.

**Comment by gbdomubpkm** [comment]
@caydey Hello. Glad to see you back. I hope you had personal success during your period of absence. Regarding your app, it is indeed essential and as a feedback after a year, I am very satisfied with it. Sometimes indeed, the compression does not launch whatever the output resolution format chosen, or it launches at a certain output format: a mystery (probably not for you). Anyway, any improvements and bug fixes will be welcome and a big thank you again !

**Comment by MeltyObserver** [comment]
any update on this issue? I like the app but I'd love to be able to use my own ffmpeg command.
maybe add it under an "advanced" label?

**Comment by MeltyObserver** [comment]
yeah this was implemented in [v1.3.0](<https://github.com/caydey/ffshare/releases/tag/v1.3.0>) (2 days  after my comment) but i guess the developer forgot to close this issue

@caydey

**Comment by caydey** [comment]
I've left the issue open because although it mentions customising ffmpeg commands, that is not the main area in the issue with other important comments that can be seen by reading past the title
